drivers/dac/dac_sam.c

```diff
@@ -25,9 +21,6 @@
 #include <zephyr/irq.h>
 LOG_MODULE_REGISTER(dac_sam, CONFIG_DAC_LOG_LEVEL);
 
-BUILD_ASSERT(IS_ENABLED(CONFIG_SOC_SERIES_SAMX7X),
-	     "Only SAMx7x series devices are currently supported.");
-
 #define DAC_CHANNEL_NO  2
```

Comment:
This line would not be correct anymore for the SAMX7X. What about changing the define depending on the SOC series and keeping the same approach with a `dac_channels` array below even if we have only one channel? That would simplify the code and reduce `#ifdef`s.

```suggestion
#if defined(SOC_SERIES_SAMX7X)
#define DAC_CHANNEL_NO  1
#else
#define DAC_CHANNEL_NO  2
#endif
```

Reply:
The value is the other way around, but I get your point.
The "difficulty" here, however is that both have 2 channels, but for the SAM4X series, a single interrupt register is used for both channels.

Users might be confused if `DAC_CHANNEL_NO` would be set to `1` in that case.

Reply:
Oops, yes, other way round.

Ok, so if they both have two channels, the single `dac_channel` to store the semaphore is a bit confusing. Should we just move the semaphore out of the struct then and call it `channel_sem` or something like that?

Reply:
There's a single interrupt register for both, I don't think we know which channel triggered the interrupt. There's a possible limitation that you can only use 1 channel at a time.

Reply:
Ok. Fine with me to keep it as you suggested.